src/components/PopupWithForm.js

```diff
@@ -20,10 +25,31 @@ class PopupWithForm extends Popup {
     });
     return inputValues;
   }
+  //this where the code for changing the save button to say "Saving..." goes
+
+  saveInfoListener() {
+    this._submitButton = this.popupForm.querySelector(".modal__button");
+    this._submitButton.textContent = "Saving...";
+  }
+
+  revertSaveButtonListener() {
+    this._submitButton = this.popupForm.querySelector(".modal__button");
+    this._submitButton.textContent = "Save";
+  }
 
   setEventListeners() {
     super.setEventListeners();
     this.popupForm.addEventListener("submit", () => {
+      this.saveInfoListener();
+      setTimeout(this._handleFormSubmit(this._getInputValues()), 5000);
+    });
+  }
+
+  updateProfileImageEventListeners() {
+    this.submitPicture = document.querySelector(".profile__image");
+    this.popupForm.addEventListener("submit", (evt) => {
+      evt.preventDefault();
```

Comment:
` evt.preventDefault();`  should be added in line 43, so the page will not reload after submitting